src/puzzle/core.cljs

```diff
@@ -22,25 +27,79 @@
     (doto loader
       (pl/spritesheet "logo" "images/spritesheet.jpg" (:piece-width @game-state) (:piece-height @game-state)))))
 
-(defn ^:private create
+(defn- display-success-text [game]
+  (let [style   {:font "40px Arial" :fill "#ff0000" :align "center"}
+        gof     (:add game)
+        world-x (get-in game [:world :center-x])
+        world-y (get-in game [:world :center-y])
+        text    (pgof/text gof
+                           world-x
+                           world-y
+                           "Congratulations! \nYou made it!"
+                           style)]
+    (pset! text :anchor (ppo/->Point 0.5 0.5))))
+
+(defn- solved? []
+  "Returns boolean for whether puzzle is solved"
+  (every? (fn [[frame-num coord]]
+            (let [board-cols (/ (:world-width @game-state) (:piece-width @game-state))]
+              (= frame-num (+ (* (coord 0) board-cols) (coord 1)))))
+          (:piece-coords @game-state)))
+
+(defn- can-move [black-coord piece-coord]
+  (or (and (= (inc (black-coord 0)) (piece-coord 0)) (= (black-coord 1) (piece-coord 1)))
+      (and (= (dec (black-coord 0)) (piece-coord 0)) (= (black-coord 1) (piece-coord 1)))
+      (and (= (inc (black-coord 1)) (piece-coord 1)) (= (black-coord 0) (piece-coord 0)))
+      (and (= (dec (black-coord 1)) (piece-coord 1)) (= (black-coord 0) (piece-coord 0)))))
+
+(defn- sprite-on-click
+  [piece event game]
+  (let [gof         (:add game)
+        tween       (pgof/tween gof piece)
+        black-coord (:black-coord @game-state)
+        black-x     (* (:piece-width @game-state) (black-coord 1))
+        black-y     (* (:piece-height @game-state) (black-coord 0))
+        piece-coord (get-in @game-state [:piece-coords (:name piece)])]
+    (when (can-move black-coord piece-coord)
+      (swap! game-state assoc :black-coord piece-coord)
+      (swap! game-state assoc-in [:piece-coords (:name piece)] black-coord)
+      (ptwn/to tween {:x black-x :y black-y} 400 "Linear" true)
+      (when (solved?) (display-success-text game)))))
+
+(defn- create-black-piece! [gof col row]
+  (let [black (pgof/sprite gof
+                           (* col (:piece-width @game-state))
+                           (* row (:piece-height @game-state)))]
+    (swap! game-state assoc :black-coord [row col])))
+
+(defn- create-puzzle-piece! [game col row sprite-key frame-number]
+  (let [gof   (:add game)
+        piece (pgof/sprite gof
+                           (* col (:piece-width @game-state))
+                           (* row (:piece-height @game-state))
+                           sprite-key
+                           frame-number)]
+    (pset! piece :name frame-number)
```

Comment:
https://github.com/dparis/phzr/blob/master/src/phzr/core.cljs#L4
Here, you can see `pset!` takes 3 arguments this( 1. object or sprite, 2. key, 3. value)